client/my-sites/hosting/github-app/index.tsx

```diff
@@ -0,0 +1,115 @@
+import { requestKeyringConnections } from 'calypso/state/sharing/keyring/actions';
+import { getKeyringServiceByName } from 'calypso/state/sharing/services/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import { GITHUB_INTEGRATION_QUERY_KEY } from './constants';
+import { DisconnectGitHubButton } from './disconnect-github-button/index';
+import {
+	GITHUB_CONNECTION_QUERY_KEY,
+	GithubConnectionData,
+	useGithubConnectionQuery,
+} from './use-github-connection-query';
+
+type Service = {
+	connect_URL: string;
+};
+
+export const GitHubAppCard = () => {
+	const queryClient = useQueryClient();
+	const siteId = useSelector( getSelectedSiteId );
+	const dispatch = useDispatch();
+	const github = useSelector( ( state ) =>
+		getKeyringServiceByName( state, 'github-app' )
+	) as Service;
+
+	const { data: connections } = useGithubConnectionQuery( siteId );
+
+	const { mutate: authorize, isPending: isAuthorizing } = useMutation< void, unknown, string >( {
+		mutationFn: async ( connectURL ) => {
+			dispatch( recordTracksEvent( 'calypso_hosting_github_authorize_click' ) );
+			await new Promise( ( resolve ) => requestExternalAccess( connectURL, resolve ) );
+			await dispatch( requestKeyringConnections() );
+		},
+		onSuccess: async () => {
+			const connectionKey = [ GITHUB_INTEGRATION_QUERY_KEY, siteId, GITHUB_CONNECTION_QUERY_KEY ];
+			await queryClient.invalidateQueries( {
+				queryKey: connectionKey,
+			} );
+
+			const authorized =
+				queryClient.getQueryData< GithubConnectionData >( connectionKey )?.connected;
+			dispatch( recordTracksEvent( 'calypso_hosting_github_authorize_complete', { authorized } ) );
+		},
+	} );
+
+	return (
+		<>
+			<QueryKeyringServices />
+			<QueryKeyringConnections />
+			<Card className="github-hosting-card">
+				<span>GitHub App</span>
+				<div
+					style={ {
+						display: 'flex',
+						flexDirection: 'column',
+						alignItems: 'flex-start',
+						gap: 16,
+						width: '100%',
+					} }
+				>
+					{ github && (
+						<Button
+							className="is-primary"
+							busy={ isAuthorizing }
+							disabled={ ! github || isAuthorizing }
+							onClick={ () => authorize( github.connect_URL ) }
+						>
+							{ translate( 'Authorize GitHub Account' ) }
+						</Button>
+					) }
+
+					{ ( connections || [] ).map( ( connection, key ) => (
+						<div key={ key } style={ { display: 'flex', alignItems: 'center', width: '100%' } }>
+							<span style={ { marginRight: 'auto' } }>{ connection.external_name }</span>
+							<span
+								style={ {
+									marginRight: 'auto',
+									maxWidth: 150,
+									fontSize: 'small',
+									overflow: 'hidden',
+									whiteSpace: 'nowrap',
+									textOverflow: 'ellipsis',
+								} }
+								title={ Object.keys( connection.installation.repositories ).join( ', ' ) }
+							>
+								Repos: { Object.keys( connection.installation.repositories ).join( ', ' ) }
```

Comment:
This is giving me a blank screen. For some reason, my authorization is not returning the repos (even though I have authorized them all). Uninstalling the app does not fix it for me. Here's the returned payload:

<img width="510" alt="image" src="https://github.com/Automattic/wp-calypso/assets/26530524/12d752c8-7f57-41d8-ae02-2fd6651bfe02">

Reply:
Can you disconnect and reconnect? The `repositories` field is added at the start.

Reply:
I can't :( The page breaks as soon as it loads.

Reply:
<img width="495" alt="Screenshot 2567-01-31 at 08 32 44" src="https://github.com/Automattic/wp-calypso/assets/6851384/5e7289bf-f157-4252-a444-4835bf1e57a8">
Ah, I forgot about that :-) Can you use this option then, @zaguiini?

Reply:
Just plug your ID in. CC @jeroenpf in case he needs to do the same.